website/react-magma-docs/src/pages/api/form.mdx

```diff
@@ -29,23 +29,128 @@ import {
 } from 'react-magma-dom';
 
 export function Example() {
+  const [state, setState] = React.useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+  });
+
+  const [errors, setErrors] = React.useState({
+    firstName: false,
+    lastName: false,
+    email: false,
+  });
+
+  const resetErrors = () => {
+    setErrors({
+      firstName: false,
+      lastName: false,
+      email: false,
+    });
+  };
+
+  const onSubmit = event => {
+    event.preventDefault();
+    resetErrors();
+
+    if (!state.firstName) {
+      setErrors(prevErrors => ({ ...prevErrors, firstName: true }));
+    }
+
+    if (!state.lastName) {
+      setErrors(prevErrors => ({ ...prevErrors, lastName: true }));
+    }
+
+    if (!state.email) {
+      setErrors(prevErrors => ({ ...prevErrors, email: true }));
+    }
+  };
+
+  const cancel = () => {
+    setState({
+      firstName: '',
+      lastName: '',
+      email: '',
+    });
+    setErrors({
+      firstName: false,
+      lastName: false,
+      email: false,
+    });
+  };
+
+  const errorMessage = React.useMemo(() => {
+    let message = '';
+
+    for (const error in errors) {
+      if (errors[error]) {
+        switch (error) {
+          case 'firstName':
+            message += '- First Name field may not be blank\n';
+            break;
+          case 'lastName':
+            message += '- Last Name field may not be blank\n';
+            break;
+          case 'email':
+            message += '- Email field may not be blank\n';
+            break;
+          default:
+            return;
+        }
+      }
+    }
+
+    return message;
+  }, [errors]);
+
   return (
     <Form
-      onSubmit={() => alert('form submitted')}
+      onSubmit={onSubmit}
       header="Form Heading"
       description="Some Form Description"
-      errorMessage="Some Form Error"
+      errorMessage={errorMessage}
```

Comment:
This looks good, the only thing is that the alert does not get focus / the message is not announced with a screenreader

Reply:
This has not been addressed yet: https://jam.dev/c/7e5f61b7-cf01-4511-acdc-409d1b8760e0

I think I expect the focus to move to the alert so that the error message can be read.